src/main/java/duke/Duke.java

```diff
@@ -79,10 +83,17 @@ public void run() {
                 String fullCommand = ui.readCommand();
                 ui.showLine();
                 Command c = CommandManager.manageCommand(fullCommand);
-                c.execute(patientTaskList, taskManager, patientManager,
-                        ui, patientTaskStorage, taskStorage, patientStorage);
-                counter.runCommandCounter(c, counterStorage, counter);
-                isExit = c.isExit();
+                if (c instanceof DukeCommand) {
+                    Command command = shortCutter.runShortCut();
+                    command.execute(patientTaskList, taskManager, patientManager,
+                            ui, patientTaskStorage, taskStorage, patientStorage);
+                } else {
+                    c.execute(patientTaskList, taskManager, patientManager,
+                            ui, patientTaskStorage, taskStorage, patientStorage);
+                    counter.runCommandCounter(c, counterStorage, counter);
+                    isExit = c.isExit();
+                }
+
```

Comment:
> Move this functional code inside your `DukeCommand.execute`, because they are functional logic code belong to your command execution.

Thanks for pointing it out. I have shifted them to duke command after your StorageManger is merged to our master.